storage/postgresql/schema.sql

```diff
@@ -0,0 +1,325 @@
+
+    CHECK (serial_number IS NULL OR serial_number != ''
+) ,
+
+    CHECK (unlock_token IS NULL OR LENGTH(unlock_token) > 0),
+
+    CHECK (authenticate != ''),
+    CHECK (token_update IS NULL OR token_update != ''),
+
+    CHECK (bootstrap_token_b64 IS NULL OR bootstrap_token_b64 != '')
+);
+CREATE INDEX serial_number ON devices (serial_number);
+
+CREATE TABLE users
+(
+    id                          VARCHAR(255) NOT NULL,
+    device_id                   VARCHAR(255) NOT NULL,
+
+    user_short_name             VARCHAR(255) NULL,
+    user_long_name              VARCHAR(255) NULL,
+
+    -- The last raw TokenUpdate for this user
+    token_update                TEXT NULL,
+    token_update_at             TIMESTAMP NULL,
+
+    -- The last raw UserAuthenticate (and optional digest) for this user
+    user_authenticate           TEXT NULL,
+    user_authenticate_at        TIMESTAMP NULL,
+    user_authenticate_digest    TEXT NULL,
+    user_authenticate_digest_at TIMESTAMP NULL,
+
+    created_at                  TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at                  TIMESTAMP DEFAULT CURRENT_TIMESTAMP, -- trigger
+
+    PRIMARY KEY (id, device_id),
+    UNIQUE (id),
+
+    FOREIGN KEY (device_id)
+        REFERENCES devices (id)
+        ON DELETE CASCADE ON UPDATE CASCADE,
+
+    CHECK (user_short_name IS NULL OR user_short_name != ''
+) ,
+    CHECK (user_long_name  IS NULL OR user_long_name  != ''),
+
+    CHECK (token_update IS NULL OR token_update != ''),
+
+    CHECK (user_authenticate        IS NULL OR user_authenticate        != ''),
+    CHECK (user_authenticate_digest IS NULL OR user_authenticate_digest != '')
+);
+
+/* This table represents enrollments which are an amalgamation of
+ * both device and user enrollments.
+ */
+CREATE TABLE enrollments
+(
+    -- The enrollment ID of this enrollment
+    id                 VARCHAR(255) NOT NULL,
+    -- The "device" enrollment ID of this enrollment. This will be
+    -- the same as the `id` field in the case of a "device" enrollment,
+    -- or will be the "parent" enrollment for a "user" enrollment.
+    device_id          VARCHAR(255) NOT NULL,
+    -- The "user" enrollment ID of this enrollment. This will be the
+    -- same as the `id` field in the case of a "user" enrollment or
+    -- NULL in the case of a device enrollment.
+    user_id            VARCHAR(255) NULL,
+
+    -- Textual representation of the type of device enrollment.
+    type               VARCHAR(31)  NOT NULL,
+
+    -- The MDM APNs push trifecta.
+    topic              VARCHAR(255) NOT NULL,
+    push_magic         VARCHAR(127) NOT NULL,
+    token_hex          VARCHAR(255) NOT NULL, -- TODO: Perhaps just CHAR(64)?
+
+    enabled            BOOLEAN      NOT NULL DEFAULT TRUE,
+    token_update_tally INTEGER      NOT NULL DEFAULT 1,
+
+    last_seen_at       TIMESTAMP    NOT NULL, -- DEFAULT CURRENT_TIMESTAMP, tests pass, but real test push error
```

Comment:
When I've just started working, I tried to test with MySQL DB. And after connecting with iPhone, I've received such log message: 
while pushing got error: handler=checkin-command msg=check-in request err=tokenupdate service: Error 1364: Field 'last_seen_at' doesn't have a default value
After forking I've changed in MySQL schema:
 last_seen_at       TIMESTAMP    NOT NULL  DEFAULT CURRENT_TIMESTAMP,
and the real device worked fine. I've forgotten about this, copied MySQL package, and started working on PostgreSQL.
After finishing, I started integration tests, but in MySQL they failed because of DEFAULT CURRENT_TIMESTAMP. 
I rolled back to the original in MySQL - and integration tests passed ok. 
I think such a situation needs some investigation :))